### DISCUSSION TITLE
is there way to define a static route/path which response a static content/page by kong self?

### DISCUSSION THREAD

**Discussion by zffocussss** [Kong/kong#8471] [Ideas and feature requests]
Hi team:
    As kong use router/service/plugin to proxy/modify requests and responses.can kong serve some requests by itself.
For example,I want to kong return "hello world to me" when the URL is /static/,without any external services.

it is a typical nginx location conf block.

/static/ {
    return "hello world";
}

I only want to use the kongplugin to implement,instead of append this block to the kong lua template file.can we do it?

**Comment by StarlightIbuki** [comment]
Sounds like you are doing some tests. Maybe this would do? https://docs.konghq.com/hub/kong-inc/request-termination/

reference: https://github.com/Kong/kong/issues/216

**Reply by zffocussss** [reply]
> 

Hi Suika:
   it is a useful function,not only for test.this is used as default backend in ingress controller(like nginx-ingress).request-termination is not ideal to do that.
